src/foam/u2/wizard/StepWizardController.js

```diff
@@ -0,0 +1,321 @@
+      of: 'foam.u2.wizard.Wizardlet',
+      postSet: function (_, n) {
+        this.setupWizardletListeners(n);
+      }
+    },
+    {
+      name: 'wizardPosition',
+      documentation: `
+        Wizardlet position
+      `,
+      class: 'FObjectProperty',
+      of: 'foam.u2.wizard.WizardPosition',
+      factory: function() {
+        return this.WizardPosition.create({
+          wizardletIndex: 0,
+          sectionIndex: 0,
+        });
+      }
+    },
+
+    {
+      name: 'nextScreen',
+      expression: function(wizardPosition) {
+        return this.nextAvailable(wizardPosition, this.positionAfter.bind(this));
+      }
+    },
+    {
+      name: 'previousScreen',
+      expression: function(wizardPosition) {
+        return this.nextAvailable(wizardPosition, this.positionBefore.bind(this));
+      }
+    },
+    {
+      name: 'visitedWizardlets',
+      class: 'FObjectArray',
+      of: 'foam.u2.wizard.Wizardlet'
+    },
+
+    // Convenience properties
+    {
+      name: 'currentWizardlet',
+      expression: function(wizardlets, wizardPosition) {
+        var current = wizardlets[wizardPosition.wizardletIndex];
+        if ( this.visitedWizardlets.indexOf(current) == -1 ) {
+          this.visitedWizardlets.push(current);
+        }
+        return current;
+      }
+    },
+    {
+      name: 'currentSection',
+      expression: function(currentWizardlet, wizardPosition$sectionIndex) {
+        return this.currentWizardlet.currentSection = currentWizardlet
+            .sections[wizardPosition$sectionIndex];
+      }
+    },
+    {
+      name: 'isLastScreen',
+      class: 'Boolean',
+      expression: function(nextScreen) {
+        return nextScreen == null;
+      }
+    },
+    {
+      name: 'canGoBack',
+      class: 'Boolean',
+      expression: function(previousScreen) {
+        return previousScreen != null;
+      }
+    },
+    {
+      name: 'canGoNext',
+      class: 'Boolean',
+      expression: function(
+        wizardPosition, nextScreen,
+        currentSection, currentWizardlet,
+        currentSection$isValid, currentWizardlet$isValid,
+        config$allowSkipping
+      ) {
+        if ( config$allowSkipping ) return true;
+        if (
+          ! nextScreen ||
+          wizardPosition.wizardletIndex != nextScreen.wizardletIndex
+        ) {
+          if ( ! currentWizardlet$isValid ) return false;
+        }
+        return currentSection$isValid;
+      }
+    },
+    {
+      name: 'submitted',
+      class: 'Boolean'
+    }
+  ],
+
+  methods: [
+    function setupWizardletListeners(wizardlets) {
+      console.log('step wizard', this, wizardlets);
+      if ( this.wsub ) this.wsub.detach();
+      this.wsub = this.FObject.create();
+
+      var wi = 0, si = 0;
+      wizardlets.forEach((w, wizardletIndex) => {
+
+        // Bind availability listener for wizardlet availability
+        var isAvailable$ = w.isAvailable$;
+        console.log(wizardletIndex, w, w.isAvailable$, isAvailable$);
```

Comment:
another console log - seems more like a debugging thing. Don't think a user will ever want to know or even understand the wizardlet object.